UFS-DA/ush/check_yaml_keys.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python3
+# compare keys between two YAML files
+import argparse
+import logging
+import os
+import yaml
+
+def check_yaml(YAMLref, YAMLtest, checkValues=False):
+    assert os.path.exists(YAMLref), f"File {YAMLref} not found."
+    assert os.path.exists(YAMLtest), f"File {YAMLtest} not found."
+    logging.basicConfig(format='%(asctime)s:%(levelname)s:%(message)s', level=logging.INFO, datefmt='%Y-%m-%d %H:%M:%S')
+    logging.info(f'Comparing {YAMLtest} against {YAMLref}...')
+    # load reference file
+    try:
+        with open(YAMLref, 'r') as YAMLref_opened:
+            ref_dict = yaml.safe_load(YAMLref_opened)
+    except Exception as e:
+        logging.error(f'Error occurred when attempting to load: {YAMLref}, error: {e}')
+    # load file to test
+    try:
+        with open(YAMLtest, 'r') as YAMLtest_opened:
+            test_dict = yaml.safe_load(YAMLtest_opened)
+    except Exception as e:
+        logging.error(f'Error occurred when attempting to load: {YAMLtest}, error: {e}')
+    # loop through top level of YAML
+    for key, value in ref_dict.items():
+        if key not in test_dict:
+            logging.error(f"Top level '{key}' not in test file.")
+        else:
+            if isinstance(value, dict):
+                compare_dict(key, value, test_dict[key], checkValues)
+            elif isinstance(value, list):
+                compare_list(key, value, test_dict[key], checkValues)
+            else:
+                if checkValues:
+                    if value != test_dict[key]:
+                        logging.warning(f"{key}:{test_dict[key]} != {value}")
```

Comment:
Can this part can be simplified by just a call to the function compare_dict you defined below?